Clarify difference between `QuantumTape` and `QuantumScript` (#5065)
PennyLaneAI/pennylane: doc/code/qml_tape.rst

    details on creating QNodes, as well as the :func:`~pennylane.qnode` decorator
    and :func:`~pennylane.QNode` constructor.

> **``QuantumTape`` versus ``QuantumScript``**

Reviewer: ```suggestion
QuantumTape versus QuantumScript
--------------------------------
```

A ``QuantumScript`` is purely a representation of a quantum circuit, and can only be constructed
via initialization. Once it is initialized, the contents should then remain immutable throughout its lifetime.

>>> ops = [qml.PauliX(0)]
>>> measurements = [qml.expval(qml.PauliZ(0))]
>>> QuantumScript(ops, measurements, shots=10)
<QuantumScript: wires=[0], params=0>

> A ``QuantumTape`` has additional queuing capabilities and also inherits from :class:`~.AnnotatedQueue`.  It's contents

Author: ```suggestion
A ``QuantumTape`` has additional queuing capabilities and also inherits from :class:`~pennylane.queuing.AnnotatedQueue`.  It's contents
```

are set on exiting the context, rather than upon initialization. Since queuing requires interaction with the global singleton
:class:`~.QueuingManager`, the ``QuantumTape`` requires a ``threading.RLock`` which complicates its use in distributed
situations.

>>> with QuantumTape(shots=10) as tape:
...     qml.PauliX(0)
...     qml.expval(qml.PauliZ(0))
>>> tape
<QuantumTape: wires=[0], params=0>

> The ``QuantumTape`` also carries around the unprocessed queue in addtion to the processed ``operations`` and ``measurements``, leading

Reviewer: ```suggestion
The ``QuantumTape`` also carries around the unprocessed queue in addition to the processed ``operations`` and ``measurements``, 
```
